gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDataset.java

```diff
@@ -316,8 +316,8 @@ protected DatasetDescriptor getDestinationDataset(FileSystem targetFs) {
     return this.destIcebergTable.getDatasetDescriptor(targetFs);
   }
 
-  private PostPublishStep createPostPublishStep(IcebergTable srcIcebergTable, IcebergTable dstIcebergTable) {
-    IcebergRegisterStep icebergRegisterStep = new IcebergRegisterStep(srcIcebergTable, dstIcebergTable);
+  private PostPublishStep createPostPublishStep(String dbName, String inputTableName, Properties properties) {
```

Comment:
seems inappropriate for the general case to serialize every single global property here.  can't we constrain only to those actually related to creating the iceberg catalogs, source and dest?

Reply:
I understand the concern here as we shouldn't need to serialize all the properties. I explored the option of doing it, but I am skeptical because we use those `properties` for providing configuration for Hadoop too which is utilized by `CatalogUtil.loadCatalog()`. And we pass the properties as-is to the configuration and it is set for an instance of Configurable: `(Configurable)catalog).setConf(hadoopConf)`. Now, it would be exhaustive to search how these hadoop configs are used by different catalog-impl. Does that make sense?

Reply:
ah yes, I see the challenge arising from the hadoop props.  these are likely to have a predictable prefix too, so we could retain only specific ones... but that might take trial-an-error, so, for now, I suggest adding a TODO documenting our preference not to serialize absolutely every prop... and yet the challenge w/ the hadoop config.

as we return to this, we should also investigate whether `Configuration` may itself already be serializable

Reply:
Added the TODO as suggested above in my latest commit!